### PR TITLE
[XZ] Add v5.2.5 with security patch and no backdoor author commits

### PR DESCRIPTION
**Pull request by mkitti**
# XZ version 5.2.5 with ZDI-22-619 patch (This pull request)
XZ v5.2.5 contains no commits from apparent backdoor author Jia Tan
https://git.tukaani.org/?p=xz.git;a=shortlog;h=refs/tags/v5.2.5

## Security patches 
We apply patch to cover CVE-2022-1271, ZDI-22-619, ZDI-CAN-16587
https://www.mail-archive.com/xz-devel@tukaani.org/msg00551.html
https://www.zerodayinitiative.com/advisories/ZDI-22-619/
https://www.cve.org/CVERecord?id=CVE-2022-1271

## Comparison to Debian and Ubuntu releases

This is the version in Debian bullseye (oldstable)
https://sources.debian.org/src/xz-utils/5.2.5-2.1~deb11u1/
This is also the version in Ubuntu 22.04LTS jammy (5.2.5-2ubuntu1)
https://packages.ubuntu.com/jammy/xz-utils
The new patch is also included in the Debian patch set:
https://sources.debian.org/src/xz-utils/5.2.5-2.1~deb11u1/debian/patches/0010-xzgrep-Fix-escaping-of-malicious-filenames-ZDI-CAN-1.patch/

See the ongoing Debian discussion in the incident bug report:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1068024

# XZ v5.2.6 (Next stable version, first with Jia Tan)

XZ v5.2.6 contains three commits from Jia Tan
https://git.tukaani.org/?p=xz.git;a=shortlog;h=refs/tags/v5.2.6
Commits:
https://git.tukaani.org/?p=xz.git;a=commit;h=e20ce2b12251a246c50fb5b7fa2204c11611b407
https://git.tukaani.org/?p=xz.git;a=commit;h=749b86c2c18ab61a07f19ec8fefc67325da97397
https://git.tukaani.org/?p=xz.git;a=commit;h=76a5a752b8467ff591dd028deb61e9bf2c274c7e
Jia Tan is added to AUTHORS here:
https://git.tukaani.org/?p=xz.git;a=commit;h=692de534fae6c38f92416793addcf5f60ffe2bf6
Conda-forge is currently locked to v5.2.6 since they yanked v5.2.9
https://anaconda.org/conda-forge/xz/files
https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/548

### CODE REVIEW

File: X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig
Comment (giordano) [inline X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig:1]: What is this for?

Comment (mkitti) [reply X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig:1]: It's the GPG signature to validate who wrote the similarly named patch.


https://tukaani.org/contact.html

Laase Collin's public key
https://tukaani.org/misc/lasse_collin_pubkey.txt

Comment (mkitti) [reply X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig:1]: OK, I will work on adding `--verify`. I suppose I also need to check in the public key?

Comment (giordano) [reply X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig:1]: You either do _something_ with that, or remove it, I frankly don't see any benefit in having an unused binary blob.

**Comment by JeffreySarnoff** [comment]
+1

File: X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig
Comment (mkitti) [reply X/XZ/bundled/patches/xzgrep-ZDI-CAN-16587.patch.sig:1]: OK, I added Laase Collin's public key and now verify both the signed tag and the signed patch in https://github.com/JuliaPackaging/Yggdrasil/pull/8396/commits/0c226b137d97aaf063a7a17398b94750b8c1f536

**Comment by giordano** [comment]
Yes, but my understanding was that @mkitti wanted to pin XZ_jll in his environments to this older version.

**Comment by imciner2** [comment]
> Yes, but my understanding was that @mkitti wanted to pin XZ_jll in his environments to this older version.

That wasn't what I was interpreting from the Slack thread. I thought he wanted it pinned for every environment.

**Comment by mkitti** [comment]
My main motivation was to make this initial step so that people can pin to this version if they are very security conscious or paranoid. Whether we yank the other later versions is above my pay grade and skill level. So far it seems probably fine, but there are a lot of commits to go through.

**Pull request merged**